### ISSUE TITLE
Unauthorized instead of Forbidden status with client_credentials

### ISSUE THREAD

**Issue by colunira** [ory/oathkeeper#387]
**Describe the bug**

I have two Rules pointing to one service which require different scopes for different endpoints (read or write). When I try to reach endpoint A, which requires "read" scope, with a token with "read" scope I get a correct response. However, when I try to reach endpoint B, which requires "write" scope, with the same token with "read" scope (which I know is proper, because I can connect to service A with it), I get 401 Unauthorized instead of expected 403 Forbidden. 

**Reproducing the bug**

1. Create a Rule and a Oauth2Client:
```
apiVersion: oathkeeper.ory.sh/v1alpha1
kind: Rule
metadata:
  name: httpbin-8m82z
  namespace: default
    kind: APIRule
    name: httpbin
spec:
  authenticators:
  - config:
      required_scope:
      - write
    handler: oauth2_introspection
  authorizer:
    handler: allow
  match:
    methods:
    - POST
    url: <http|https>://httpbin.kyma.local</post>
  upstream:
    url: http://httpbin.default.svc.cluster.local:8000
```
```
apiVersion: hydra.ory.sh/v1alpha1
kind: OAuth2Client
metadata:
  name: test-client
  namespace: default
spec:
  grantTypes:
  - "client_credentials"
  scope: "read write"
  secretName: test-secret
```

2. Fetch the access token from Hydra with scope read using data from connected to OAuth2Client secret:

```curl -X POST "https://{ADDRESS}/oauth2/token" -H "Authorization: Basic {CREDENTIALS}" -F "grant_type=client_credentials" -F "scope=write"```

3. Call exposed service:

```curl -X POST "https://httpbin.kyma.local/post" -H "Authorization: Bearer {ACCESS_TOKEN}"```

*Oathkeeper logs*

```
time="2020-03-25T11:30:39Z" level=info msg="started handling request" method=GET remote="127.0.0.1:49626" request=/headers request_id=48e57029-c9cc-4a45-9564-6afc6e0f8bef
time="2020-03-25T11:30:39Z" level=warning msg="The authentication handler encountered an error" authentication_handler=oauth2_introspection error="Access credentials are invalid" granted=false http_host=httpbin.kyma.local http_method=GET http_url="http://httpbin.kyma.local/headers" http_user_agent=curl/7.66.0 reason_id=authentication_handler_error rule_id=httpbin-kh4vn.default
time="2020-03-25T11:30:39Z" level=warning msg="Access request denied" error="Access credentials are invalid" granted=false http_host=httpbin.kyma.local http_method=GET http_url="http://httpbin.kyma.local/headers" http_user_agent=curl/7.66.0
time="2020-03-25T11:30:39Z" level=error msg="An error occurred while handling a request" code=401 debug= details="map[]" error="The request could not be authorized" reason= request-id=48e57029-c9cc-4a45-9564-6afc6e0f8bef status=401 trace="Stack trace: \ngithub.com/ory/oathkeeper/pipeline/authn.(*AuthenticatorOAuth2Introspection).Authenticate\n\t/home/circleci/project/pipeline/authn/authenticator_oauth2_introspection.go:109\ngithub.com/ory/oathkeeper/proxy.(*RequestHandler).HandleRequest\n\t/home/circleci/project/proxy/request_handler.go:213\ngithub.com/ory/oathkeeper/proxy.(*Proxy).Director\n\t/home/circleci/project/proxy/proxy.go:134\nnet/http/httputil.(*ReverseProxy).ServeHTTP\n\t/usr/local/go/src/net/http/httputil/reverseproxy.go:206\ngithub.com/urfave/negroni.Wrap.func1\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:46\ngithub.com/urfave/negroni.HandlerFunc.ServeHTTP\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:29\ngithub.com/urfave/negroni.middleware.ServeHTTP\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38\ngithub.com/ory/x/reqlog.(*Middleware).ServeHTTP\n\t/go/pkg/mod/github.com/ory/x@v0.0.91/reqlog/middleware.go:140\ngithub.com/urfave/negroni.middleware.ServeHTTP\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38\ngithub.com/ory/x/metricsx.(*Service).ServeHTTP\n\t/go/pkg/mod/github.com/ory/x@v0.0.91/metricsx/middleware.go:261\ngithub.com/urfave/negroni.middleware.ServeHTTP\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38\ngithub.com/urfave/negroni.(*Negroni).ServeHTTP\n\t/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:96\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2802\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1890\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357" writer=JSON
time="2020-03-25T11:30:39Z" level=info msg="completed handling request" measure#oathkeeper-proxy.latency=11369609 method=GET remote="127.0.0.1:49626" request=/headers request_id=48e57029-c9cc-4a45-9564-6afc6e0f8bef status=401 text_status=Unauthorized took=11.369609ms
```

**Expected behavior**

Get a 403 Forbidden response when I connect to an endpoint with proper access token, but without required scopes.

**Environment**

minikube 1.7.3, Kubernetes 1.16.3

**Comment by aeneasr** [comment]
Yeah, I guess we could return a 403 there. 401 is being returned because the error happens in the authenticator / authentication process. I guess it's a bit ambiguous what to return there as both 401 and 403 would be valid imo. However, 403 also makes sense and we're open to contributions! :)

**Comment by flaviuvadan** [comment]
I was looking through the stack trace reported by the log and it lead me to the introspection package. I found
```go
if !i.Active {
	return errors.WithStack(helper.ErrUnauthorized.WithReason("Access token i says token is not active"))
}
```
which seems to be the part of `Authenticate` that returns a 401 instead of a 403. It's a simple change. I also found `The Token is considered valid if the Introspection response is HTTP 200 OK and includes {"active":true} in the response payload. The subject is extracted from the username field.` in the [docs](https://www.ory.sh/oathkeeper/docs/pipeline/authn/#access-rule-example-5). I'd love to submit a PR, but it'd be my first one so if there's anything I'm missing, I'd be happy to know!
